### PR TITLE
Change versioning to setuptools_scm_git_archive to sync with tags.

### PR DESCRIPTION
**Pull request by erykoff**
I recently learned from @beckermr for this one simple trick to get the versioning to sync with git tags.  Now versions will be auto-generated with the git ref, and if we push a new tag then it gets incorporated into the package automatically (and a release is made with that tag).

### CODE REVIEW

**Comment by erykoff** [comment]
For a bit more info, I was just copying what was done here: https://github.com/beckermr/pizza-cutter and looking at the docs here: https://pypi.org/project/setuptools-scm-git-archive/

**Review by beckermr — APPROVED** [review]

File: healsparse/healSparseMap.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def read(cls, filename, nside_coverage=None, pixels=None, header=False,
                     row = fits.read_ext_data(1, row_range=[0, 1])
                     dtype = row[0][0][0].dtype.type
 
-            healpix_map = hp.read_map(filename, nest=True, verbose=False, dtype=dtype)
+            healpix_map = hp.read_map(filename, nest=True, dtype=dtype)
```
Comment (fjaviersanchez) [inline healsparse/healSparseMap.py:167]: Was this change intentional?

**Review by fjaviersanchez — APPROVED** [review]
Thanks @erykoff! Looks good to me

File: healsparse/healSparseMap.py (hunk 1 above)
Comment (erykoff) [reply healsparse/healSparseMap.py:167]: Yes, this was a separate commit.  I removed it because the `verbose` keyword has been deprecated and does nothing.
